crates/shielded_token/src/conversion.rs

```diff
@@ -436,33 +441,39 @@ where
                 // Express the inflation reward in real terms, that is, with
                 // respect to the native asset in the zeroth
                 // epoch
-                let real_reward = ((Uint::from(reward.0)
-                    * Uint::from(ref_inflation))
-                    / normed_inflation)
-                    .try_into()
-                    .unwrap_or_else(|_| {
-                        tracing::warn!(
-                            "MASP reward for {} assumed to be 0 because the \
-                             computed value is too large. Please check the \
-                             inflation parameters.",
-                            token
-                        );
-                        0u128
-                    });
+                let reward_uint = Uint::from(reward);
+                let ref_inflation_uint = Uint::from(ref_inflation);
+                let inflation_uint = Uint::from(normed_inflation);
+                let real_reward = checked!(
+                    (reward_uint * ref_inflation_uint) / inflation_uint
+                )?
+                .try_into()
+                .unwrap_or_else(|_| {
+                    tracing::warn!(
+                        "MASP reward for {} assumed to be 0 because the \
+                         computed value is too large. Please check the \
+                         inflation parameters.",
+                        token
+                    );
+                    0u128
+                });
                 // The conversion is computed such that if consecutive
                 // conversions are added together, the
                 // intermediate tokens cancel/ telescope out
-                let reward_i128 =
-                    i128::try_from(reward.1).into_storage_result()?;
+                let precision_i128 =
+                    i128::try_from(precision).into_storage_result()?;
+                let real_reward_i128 =
+                    i128::try_from(real_reward).into_storage_result()?;
                 current_convs.insert(
                     (token.clone(), denom, digit),
-                    (MaspAmount::from_pair(old_asset, -reward_i128)
-                        + MaspAmount::from_pair(new_asset, reward_i128)
-                        + MaspAmount::from_pair(
-                            reward_assets[digit as usize],
-                            i128::try_from(real_reward)
-                                .into_storage_result()?,
-                        ))
+                    checked!(
+                        MaspAmount::from_pair(old_asset, -precision_i128)
```

Comment:
Should we use the same `checked_neg` we use above?

Reply:
when it's inside `checked!` it gets automatically converted to it by the macro. I haven't done it in the above one cause it's chained with int conversion